scripts/SAut_F_4.jl

```diff
@@ -0,0 +1,168 @@
+using LinearAlgebra
+ENV["JULIA_NUM_THREADS"] = Sys.CPU_THREADS÷2
+LinearAlgebra.BLAS.set_num_threads(Sys.CPU_THREADS÷2)
+
+using LowCohomologySOS
+using Groups
+using JuMP
+using SymbolicWedderburn
+using PermutationGroups
+include(joinpath(@__DIR__, "optimizers.jl"))
+include(joinpath(@__DIR__, "utils.jl"))
+
+function determine_transvection(
+    g::Groups.GroupElement # has to be a generating transvection or its inverse
+)
+    SAutFn = parent(g)
+    Fn = SAutFn.group
+    Fn_gens = Groups.gens(Fn)
+    g_id, g_i, g_j, g_inv = 0, 0, 0, 0
+    for k in eachindex(Fn_gens)
+        if g(Fn_gens[k]) != Fn_gens[k]
+            g_i = k
+            for l in eachindex(Fn_gens)
+                if g(Fn_gens[k]) == Fn_gens[k]*Fn_gens[l]
+                    g_id = :ϱ
+                    g_j = l
+                    g_inv = false
+                    break
+                elseif g(Fn_gens[k]) == Fn_gens[k]*Fn_gens[l]^(-1)
+                    g_id = :ϱ
+                    g_j = l
+                    g_inv = true
+                    break
+                elseif g(Fn_gens[k]) == Fn_gens[l]*Fn_gens[k]
+                    g_id = :λ
+                    g_j = l
+                    g_inv = false
+                    break
+                elseif g(Fn_gens[k]) == Fn_gens[l]^(-1)*Fn_gens[k]
+                    g_id = :λ
+                    g_j = l
+                    g_inv = true
+                    break
+                end
+            end
+            break    
+        end
+    end
+
+    return Groups.Transvection(g_id, g_i, g_j, g_inv)
+end
```

Comment:
`Alphabet` defines a bijection between 1:n and letters of alphabet, ie.
```
A = Alphabet(...)
l = A[1] # → the first letter of A
A[l] # the index of l, i.e. 1
```
Actually it is also possible to do `A[-1]` to obtain __the inverse__ of the first letter (if it exists).

```julia
function determine_transvection(g::Automorphism)
    @assert parent(g) isa Automorphism{<:FreeGroup}
    @assert length(word(g)) = 1
    A = alphabet(parent(g))
    return A[first(word(g))]
end

Reply:
hmm? why resolved when the old `determine_transvection` is still there?

Reply:
sorry, forgot to change it, now shall be ok ;)